packages/explorer/src/app/(explorer)/[chainName]/worlds/[worldAddress]/interact/content/useEnsAddress.ts

```diff
@@ -2,34 +2,42 @@ import { createPublicClient, http } from "viem";
 import { getEnsAddress } from "viem/actions";
 import { mainnet } from "viem/chains";
 import { normalize } from "viem/ens";
-import { useQuery } from "@tanstack/react-query";
+import { skipToken, useQuery } from "@tanstack/react-query";
 
 const mainnetClient = createPublicClient({
   chain: mainnet,
   transport: http(),
 });
 
 function isValidEnsName(name: string): boolean {
-  return name.includes(".");
+  if (!name.includes(".")) {
+    return false;
+  }
+
+  const parts = name.split(".");
+  return parts.length >= 2 && !!parts[1] && parts[1].length > 0;
```

Comment:
easier with regex imo
```ts
function isValidEnsName(name: string): boolean {
  return /\w\.\w/.test(name);
}
```

Reply:
I still think this should be inlined though, because this isn't actually telling you if it's a valid ENS, just that the string is possibly an ENS

Reply:
oh amazing, yeah much better